plugin-bazel/src/main/kotlin/org/jetbrains/bazel/symbols/BazelTargetIndex.kt

```
@@ -0,0 +1,192 @@
+     * Get targets in a specific package
+     */
+    fun getTargetsInPackage(packagePath: String, project: Project): List<BazelTargetInfo> {
+      val targets = mutableListOf<BazelTargetInfo>()
+      
+      FileBasedIndex.getInstance().processAllKeys(INDEX_ID, { targetName ->
+        val targetInfos = getTargetsByName(targetName, project)
+        targets.addAll(targetInfos.filter { it.packagePath == packagePath })
+        true
+      }, GlobalSearchScope.projectScope(project), null)
+      
+      return targets
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)